### ISSUE TITLE
Pick up external changes during the build with vfs retention

### ISSUE THREAD

**Issue by wolfs** [gradle/gradle#12163]
We currently on watch the file system in between builds, so if some changes are made during the build we won't pick them up. For correct builds, this is necessary since the next build will not detect the change if it happens during the build before the watchers are registered.

Depends on #11901.

---
cc: @gradle/build-cache

**Comment by wolfs** [comment]
Duplicate of #11837
